controller/server/main.go

```diff
@@ -14,24 +14,101 @@
 package main
 
 import (
+	"context"
 	"flag"
 	"fmt"
-	"log"
 	"net"
+	"path/filepath"
 
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/util/homedir"
+
+	"github.com/google/kne/deploy"
 	cpb "github.com/google/kne/proto/controller"
-	"google.golang.org/grpc"
+
 	"google.golang.org/grpc/credentials/alts"
+
+	"google.golang.org/grpc"
+)
+
+const (
+	defaultMetallbManifestDir = "../../manifests/metallb"
+	defaultMeshnetManifestDir = "../../manifests/meshnet/base"
 )
 
 var (
-	port = flag.Int("port", 50051, "Controller server port")
+	port           = flag.Int("port", 50051, "Controller server port")
+	defaultKubeCfg = ""
 )
 
 type server struct {
 	cpb.UnimplementedTopologyManagerServer
 }
 
+func getDeployment(req *cpb.CreateClusterRequest) (*deploy.Deployment, error) {
+	d := &deploy.Deployment{}
+	switch kind := req.ClusterSpec.(type) {
+	case *cpb.CreateClusterRequest_Kind:
+		k := &deploy.KindSpec{}
+		k.Name = req.GetKind().Name
+		k.Recycle = req.GetKind().Recycle
+		k.Version = req.GetKind().Version
+		k.Image = req.GetKind().Image
+		d.Cluster = k
+	default:
+		return nil, fmt.Errorf("cluster type not supported: %T", kind)
+	}
+	switch metallb := req.IngressSpec.(type) {
+	case *cpb.CreateClusterRequest_Metallb:
+		l := &deploy.MetalLBSpec{}
+		if req.GetMetallb().ManifestDir == "" {
+			l.ManifestDir = defaultMetallbManifestDir
+		} else {
+			l.ManifestDir = req.GetMetallb().ManifestDir
```

Comment:
There should be a check here that the given manifestDir exists and if so translate it to an absolute path, if not throw an error right away

Reply:
Done